hazelcast/src/main/java/com/hazelcast/spi/tenantcontrol/TenantControl.java

```diff
@@ -33,7 +33,7 @@
  * @author lprimak
  */
 @Beta
-public interface TenantControl extends Serializable {
+public interface TenantControl extends DataSerializable {
```

Comment:
no reason to force `DataSerializable` here. The actual implementation can be `IdentifiedDataSerializable` or plain `DataSerializable` and Hazelcast serializers will prefer that serialization type over plain `Serializable`. This is similar to how we handle for example `EntryProcessor` and other interfaces meant for user customization.

Reply:
it's better to use DataSerializable here for implementor's sake. This makes it easier, otherwise, instead of implementing readData/writeData() the user is forced to deal with defaultRedObject() and those subtleties